### PR TITLE
Upgrade package to ESM

### PR DESCRIPTION
**Pull request by benja**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 node_modules
-yarn.lock
```
Comment (benja) [inline .gitignore:2]: Can't run security fixes if we don't have a lock-file pushed

File: index.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
-const { join } = require('path');
+import { join } from "path";
 
-module.exports = {
-  iconsPath: join(__dirname, 'dist'),
+// Since __dirname no longer exists with ESM, https://stackoverflow.com/a/50052194
+import { dirname } from "path";
+import { fileURLToPath } from "url";
+export const __dirname = dirname(fileURLToPath(import.meta.url));
```
Comment (benja) [inline index.js:6]: `__dirname` doesn't work as expected in esm, so we need this

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@fabric-ds/icons",
   "type": "module",
-  "version": "0.3.12",
+  "version": "0.3.14",
```
Comment (benja) [inline package.json:4]: already did a publish

**Comment by pearofducks** [comment]
Missing `type: module` from package.json?

package-lock.json shouldn't be checked in - Fabric uses yarn :)

**Comment by benja** [comment]
> Missing type: module from package.json?

Uh, no it seems to be [here](https://github.com/fabric-ds/icons/pull/4/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R3)?

**Comment by pearofducks** [comment]
Ah I was looking for it in the green changes, I'm an idiot - lol

**Comment by benja** [comment]
Not sure why it wasn't marked in green 😆

**Review by pearofducks — APPROVED** [review]
once the package-lock.json is dead this looks great! thanks again!

File: index.js (hunk 2 above)
Comment (trygve-lie) [reply index.js:6]: The recommended way to do this by the node.js doc is as follow:

```js
const absolute_filepath = new URL('../a/path/to/a/file.js', import.meta.url);
```

The above will give you a URL object which node.js file APIs understand but if you need a absolute path to a file as a string one can do as so:

```js
const absolute_filepath = new URL('../a/path/to/a/file.js', import.meta.url).pathname;
```

**Pull request merged**